From: Takashi Iwai <tiwai@suse.de>
Subject: [PATCH RFC v2 4/4] firmware: Install firmware signature files automatically
Date: Thu,  8 Nov 2012 18:35:09 +0100
Patch-mainline: Never, firmware signing rejected upstream
References: fate#314574
Target: SLE-11 SP3

... when CONFIG_FIRMWARE_SIG is set.

Signed-off-by: Takashi Iwai <tiwai@suse.de>
Acked-by: Lee, Chun-Yi <jlee@suse.com>
[2013-01-28: Add dependency on CONFIG_MODULE_SIG_ALL --mmarek]
---
 Makefile                |    7 +++++++
 scripts/Makefile.fwinst |   19 +++++++++++++++++--
 2 files changed, 24 insertions(+), 2 deletions(-)

--- a/Makefile
+++ b/Makefile
@@ -748,6 +748,13 @@ mod_sign_cmd = true
 endif
 export mod_sign_cmd
 
+fw_sign_cmd = true
+ifeq ($(CONFIG_FIRMWARE_SIG),y)
+ifdef CONFIG_MODULE_SIG_ALL
+fw_sign_cmd = perl $(srctree)/scripts/sign-file -f $(CONFIG_MODULE_SIG_HASH) $(MODSECKEY) $(MODPUBKEY)
+endif
+endif
+export fw_sign_cmd
 
 ifeq ($(KBUILD_EXTMOD),)
 core-y		+= kernel/ mm/ fs/ ipc/ security/ crypto/ block/
--- a/scripts/Makefile.fwinst
+++ b/scripts/Makefile.fwinst
@@ -37,6 +37,21 @@ installed-mod-fw := $(addprefix $(INSTAL
 
 installed-fw := $(addprefix $(INSTALL_FW_PATH)/,$(fw-shipped-all))
 
+installed-fw-sig :=
+installed-mod-fw-sig :=
+ifdef CONFIG_MODULE_SIG_ALL
+ifdef CONFIG_FIRMWARE_SIG
+installed-fw-sig := $(patsubst %,%.sig, $(installed-fw))
+installed-mod-fw-sig := $(patsubst %,%.sig, $(installed-mod-fw))
+endif
+endif
+
+quiet_cmd_fwsig = FWSIG $@
+      cmd_fwsig = $(fw_sign_cmd) $(patsubst %.sig,%,$@) $@
+
+%.sig: %
+	$(call cmd,fwsig)
+
 quiet_cmd_install = INSTALL $(subst $(srctree)/,,$@)
       cmd_install = mkdir -p $(@D); $(INSTALL) -m0644 $< $@
 
@@ -47,9 +62,9 @@ PHONY +=  __fw_install __fw_modinst FORC
 
 .PHONY: $(PHONY)
 
-__fw_install: $(installed-fw)
+__fw_install: $(installed-fw) $(installed-fw-sig)
 
-__fw_modinst: $(installed-mod-fw)
+__fw_modinst: $(installed-mod-fw) $(installed-mod-fw-sig)
 	@:
 
 __fw_modbuild: $(addprefix $(obj)/,$(mod-fw))
